### PR TITLE
Add NVIDIA Compiler Support

### PR DESCRIPTION
**Pull request by gha3mi**
Hello,

This PR refers to #52.

Within this PR, NVIDIA compiler support has been added. Please note that there is no NVIDIA compiler available for Windows and Mac.

Thanks,
Ali

### CODE REVIEW

**Comment by wpbonelli** [comment]
Thanks @gha3mi, looks good to me. Would be good to get the go-ahead from @awvwgk

**Review by wpbonelli — APPROVED** [review]

File: setup-fortran.sh
Comment (gha3mi) [reply setup-fortran.sh:580]: Version 20.7 is the oldest version that can be found at https://developer.nvidia.com/nvidia-hpc-sdk-releases and is available on APT.

File: README.md
Comment (gha3mi) [reply README.md:45]: I believe users don't need to search for a solution themselves. A straightforward and functional usage guide could be beneficial. During my initial use, I simply copied and pasted the usage provided in the README into my repositories, and I found it simple and easy to use.

File: setup-fortran.sh
Comment (gha3mi) [reply setup-fortran.sh:580]: https://developer.download.nvidia.com/hpc-sdk/ubuntu/amd64/

Comment (wpbonelli) [reply setup-fortran.sh:580]: Is there any limitation related to the specific versions you've chosen to add to the full test matrix? If not, I see no reason not to test/support them all, from 20.7 up to current

Comment (gha3mi) [reply setup-fortran.sh:580]: It appears that versions 23.1, 22.7, 22.2, and possibly 21.1 encounter issues due to large data size during installation on the server.

Versions 20.7 and 20.9 are facing difficulties in locating the packages, requiring debugging to find the problem.

All other versions have been successfully installed.

The following are the outcomes of my tests on a private test repository:

**23.1**: 

```
dpkg: error processing archive /var/cache/apt/archives/nvhpc-23-1_23.1_amd64.deb (--unpack):
 cannot copy extracted data for './opt/nvidia/hpc_sdk/Linux_x86_64/23.1/profilers/Nsight_Systems/host-linux-x64/Mesa/libGL.so.1.5.0' to '/opt/nvidia/hpc_sdk/Linux_x86_64/23.1/profilers/Nsight_Systems/host-linux-x64/Mesa/libGL.so.1.5.0.dpkg-new': failed to write (No space left on device)
No apport report written because the error message indicates a disk full error
```

**22.7**:
```
dpkg: error processing archive /var/cache/apt/archives/nvhpc-22-7_22.7_amd64.deb (--unpack):
 cannot copy extracted data for './opt/nvidia/hpc_sdk/Linux_x86_64/22.7/profilers/Nsight_Compute/host/linux-desktop-glibc_2_11_3-x64/Plugins/libLinuxPPCPlatformPlugin.so' to '/opt/nvidia/hpc_sdk/Linux_x86_64/22.7/profilers/Nsight_Compute/host/linux-desktop-glibc_2_11_3-x64/Plugins/libLinuxPPCPlatformPlugin.so.dpkg-new': failed to write (No space left on device)
No apport report written because the error message indicates a disk full error
```

**22.2**:
```
cannot copy extracted data for './opt/nvidia/hpc_sdk/Linux_x86_64/22.2/profilers/Nsight_Systems/pkg/target-linux-x64/libcupti.so.11.2' to '/opt/nvidia/hpc_sdk/Linux_x86_64/22.2/profilers/Nsight_Systems/pkg/target-linux-x64/libcupti.so.11.2.dpkg-new': failed to write (No space left on device)
```

**21.1**:
```
File has unexpected size (2115251386 != 2115276192). Mirror sync in progress?
```

**20.9**:
```
E: Unable to locate package nvhpc-20-9
```

**20.7**:
```
E: Unable to locate package nvhpc-20-7
```

Comment (wpbonelli) [reply setup-fortran.sh:580]: Sounds good. We can still test them all in the full matrix, the failing ones will be reported as such when the bot updates the readme.

**Review by awvwgk — APPROVED** [review]
Thanks for sharing. This is a great addition.

File: .github/workflows/test.yml
Comment (wpbonelli) [inline .github/workflows/test.yml:140]: @gha3mi  can you update this to all versions before we merge?

Comment (gha3mi) [reply .github/workflows/test.yml:140]: https://github.com/fortran-lang/setup-fortran/pull/53/commits/7b7f896da30a7ac703543029453d2b0cb520d735 includes all available versions.

**Pull request merged**
